src/main/java/com/pinterest/secor/parser/MessageParser.java

```diff
@@ -60,7 +60,7 @@ public ParsedMessage parse(Message message) throws Exception {
         String[] partitions = extractPartitions(message);
```

Comment:
In the old way, the extractPartitions(message) will extract the timestamp from the message and generate the partition array from that timestamp.

How does your newly extracted kafka timestamp used in that situation then?